sdk/src/message.rs

```diff
@@ -322,6 +322,159 @@ impl Message {
         }
         (writable_keys, readonly_keys)
     }
+
+    fn append_u16(buf: &mut Vec<u8>, data: u16) {
+        let start = buf.len();
```

Comment:
How about using `byteorder` here?

Reply:
Actually I wrote it first with byteorder and then switched to this. I ran into some compatibility with byteorder when in BPF mode, and I figured it was basically the same or very similar logic and without a dependency on `byteorder`.

Reply:
Maybe we should formalize these methods and make them available to BPF programs so they can do their own serialization efficiently

Reply:
I can move it to some kind of `serialize_utils` module if it makes sense.